### ISSUE TITLE
Triplet loss for training

### ISSUE THREAD

**Issue by davidjaw** [davidjaw/NTU-DLCV-final-Face-Recognition#8]
Basically, it's a training trick that I came up with while the validation accuracy stuck around 82%.

**This trick is only utilized in the Teacher Network**

The flowchart looks like this:
![flowchart 5](https://user-images.githubusercontent.com/6800429/41893279-c1262678-794d-11e8-8d61-426f8ca4a7af.png)

An additional embedding layer is added after the CNNs to apply the [Triplet Loss](https://arxiv.org/abs/1503.03832), but the weights of the additional layer won't be saved since I didn't look into the [code](https://github.com/omoindrot/tensorflow-triplet-loss) for how they produces prediction (lol).
In other words, this triplet loss can be viewed as a multi-task optimization trick.

If the embedding layer can produce prediction, maybe we can use the ensemble architecture.
